index.js

```diff
@@ -32,3 +32,22 @@ module.exports = fn => {
 		};
 	};
 };
+
+module.exports.proxy = fn => {
+	const cache = [];
```

Comment:
Global cache feels like a bit of an overkill. I was thinking just a closure cache like the existing implementation:

```js
module.exports.proxy = fn => {
	return id => {
		let mod;

		const handler = {
			get: (target, property) => {
				mod = lazy(mod, fn, id);
				return mod[property];
			},
			apply: (target, thisArg, argumentsList) => {
				mod = lazy(mod, fn, id);
				return mod.apply(thisArg, argumentsList);
			}
		};

		return new Proxy(() => {}, handler);
	};
};
```

Reply:
Just curious, what are the benefits of using `Reflect`? You mentioned safety and clarity, but how is it safer? And clarity it subjective, I would find what's here "more clear," but mostly because of my before-now unfamiliarness of `Reflect`.

Reply:
In case the module exports an `apply` method or something. The old way to safe-guard was to use `Function.prototype.apply.call(f, obj, args)`, but now we have `Reflect.apply`, which is more readable.

Reply:
Cool, thanks